metadata-ingestion/src/datahub/ingestion/source/powerbi/powerbi.py

```diff
@@ -1179,11 +1203,8 @@ def get_workspace_workunit(
             ):
                 yield work_unit
 
-        for dataset in self.mapper.processed_datasets:
-            if self.source_config.extract_datasets_to_containers:
-                dataset_workunits = self.mapper.generate_container_for_dataset(dataset)
-                for workunit in dataset_workunits:
-                    yield workunit
+        yield from self.extract_datasets_as_containers()
```

Comment:
i'd like to keep the `self.source_config.extract_datasets_to_containers` check here

Reply:
done